src/pages/Script/index.tsx

```diff
@@ -0,0 +1,235 @@
+        </span>
+      ),
+    },
+    {
+      title: i18n.t('scripts.capacity_of_deployed_cells'),
+      content: capacityOfDeployedCells,
+    },
+    {
+      title: i18n.t('scripts.capacity_of_deployed_cells'),
+      content: capacityOfReferringCells,
+    },
+  ]
+
+  return items
+}
+
+const ScriptsTitleOverview = ({ scriptInfo }: { scriptInfo: ScriptInfo }) => {
+  const dispatch = useDispatch()
+  return (
+    <ScriptsTitleOverviewPanel>
+      <OverviewCard items={getScriptInfo(scriptInfo, dispatch)} hideShadow />
+    </ScriptsTitleOverviewPanel>
+  )
+}
+
+export interface ScriptInfo {
+  scriptName: string
+  scriptType: string
+  typeId: number
+  codeHash: string
+  hashType: string
+  capacityOfDeployedCells: string
+  capacityOfReferringCells: string
+  ckbTransactions?: {
+    total?: number
+    page?: number
+    size?: number
+  }
+  deployedCells?: {
+    total?: number
+    page?: number
+    size?: number
+  }
+  referringCells?: {
+    total?: number
+    page?: number
+    size?: number
+  }
+}
+
+const initScriptInfo: ScriptInfo = {
+  scriptName: '',
+  scriptType: '',
+  typeId: 0,
+  codeHash: '',
+  hashType: '',
+  capacityOfDeployedCells: '',
+  capacityOfReferringCells: '',
+  ckbTransactions: {
+    total: 0,
+    page: 1,
+    size: 10,
+  },
+  deployedCells: {
+    total: 0,
+    page: 1,
+    size: 10,
+  },
+  referringCells: {
+    total: 0,
+    page: 1,
+    size: 10,
+  },
+}
+
+export type ScriptTabType = 'transactions' | 'deployed_cells' | 'referring_cells' | undefined
+
+export const ScriptPage = () => {
+  const history = useHistory()
+  const { search } = useLocation()
+  const { codeHash, hashType, tab } = useParams<{ codeHash: string; hashType: string; tab: ScriptTabType }>()
+  const anchor = (tab ?? 'transactions') as ScriptTabType
+  const [activeTab, setActiveTab] = useState<ScriptTabType>(anchor)
+  const searchParams = new URLSearchParams(search)
+  const currentPage = parsePageNumber(searchParams.get('page'), PageParams.PageNo)
+  const pageSize = parsePageNumber(searchParams.get('size'), PageParams.PageSize)
+  const [scriptInfo, setScriptInfo] = useState<ScriptInfo>(initScriptInfo)
+
+  useEffect(() => {
+    if (anchor === 'deployed_cells') {
+      setActiveTab('deployed_cells')
+    } else if (anchor === 'referring_cells') {
+      setActiveTab('referring_cells')
+    } else {
+      setActiveTab('transactions')
+    }
+  }, [anchor, codeHash, currentPage, hashType, pageSize, scriptInfo, tab])
+
+  return (
+    <Content>
+      <ScriptContentPanel className="container">
+        <HashCardPanel isColumn={false}>
+          <ScriptsTitleOverview scriptInfo={scriptInfo} />
+        </HashCardPanel>
+        <ScriptTabs
+          activeKey={activeTab}
+          onTabClick={key => {
+            if (key === 'deployed_cells') {
+              history.push(
+                `/script/${codeHash}/${hashType}/deployed_cells?page=${scriptInfo!.deployedCells!.page}&size=${
```

Comment:
`scriptInfo` will not be empty, and the `!` should be removed.

Properties such as `deployedCells` should also not be asserted by `!`, it should be `if (key === 'deployed_cells' && scriptInfo.deployedCells ! = null)` or something like that.

Reply:
> `scriptInfo` will not be empty, and the `!` should be removed.
> 
> Properties such as `deployedCells` should also not be asserted by `!`, it should be `if (key === 'deployed_cells' && scriptInfo.deployedCells ! = null)` or something like that.

fixed